nucypher/characters/control/specifications/fields/treasuremap.py

```diff
@@ -40,13 +35,19 @@ def _deserialize(self, value, attr, data, **kwargs):
             raise InvalidInputData(f"Could not parse {self.name}: {e}")
 
     def _validate(self, value):
-
-        splitter = BytestringSplitter(Signature,
-                                  (bytes, HRAC_LENGTH),  # hrac
-                                  (UmbralMessageKit, VariableLengthBytestring)
-                                  )  # TODO: USe the one from TMap
-        try:
-            signature, hrac, tmap_message_kit = splitter(value)
-            return True
-        except InvalidNativeDataTypes as e:
-            raise InvalidInputData(f"Could not parse {self.name}: {e}")
+        from nucypher.policy.collections import SignedTreasureMap
+        from nucypher.policy.collections import TreasureMap as TreasureMapBase
+
+        # TODO unsure which is which is there a way for one splitter to work in both cases?
```

Comment:
This should be easier with the `get_splitter()` in #2719.

Although looking at a slightly bigger picture, how come we don't know at this point whether it's a signed treasure map or a not signed one? Shouldn't there rather be a separate `class SignedTreasureMap` definition here?

Reply:
`_validate()` is called as part of broader `deserialize()` function for Fields, and is passed the result of `_deserialize()` (function above it). This means that the `value` parameter passed to `_validate()` is in bytes, and therefore it's not clear to me that we can differentiate whether the bytes are expected to be (note not what we receive, but rather what we expect to receive) i.e. a signed tmap or a federated tmap

In practice, it will probably be just a signed treasure map since Porter is intended for decentralized workflow, but I've found accommodating the federated workflow to be good for testing.

Reply:
What I mean is that, for example, we don't try to decide from a stream of bytes whether it's a public key or a signature. We already know what it should be, and if it's not that, we raise and exception. Why doesn't it work the same for treasure maps? 

I am not familiar with this part of the code. Is it because this validator is used on something the user is passed via CLI? Something like `nucypher bob --treasure-map ...`. So it is not a good idea to make a separate `--signer-treasure-map` option, and even though we know whether we in a federated or a blockchain context, there is currently no mechanism to pass that information through `deserialize()`?

Reply:
It would be something like this:

```bash
$ curl <PORTER_URI>/publish_treasure_map -X GET -H "Content-Type: application/json" -d ' {"treasure_map": "Qld7S8sbKFCv2B8KxfJo4oxiTOjZ4VPyqTK5K1xK6DND6TbLg2hvlGaMV69aiiC5QfadB82w/5q1Sw+SNFHN2esWgAbs38QuUVUGCzDoWzQAAAGIAuhw12ZiPMNV8LaeWV8uUN+au2HGOjWilqtKsaP9fmnLAzFiTUAu9/VCxOLOQE88BPoWk1H7OxRLDEhnBVYyflpifKbOYItwLLTtWYVFRY90LtNSAzS8d3vNH4c3SHSZwYsCKY+5LvJ68GD0CqhydSxCcGckh0unttHrYGSOQsURUI4AAAEBsSMlukjA1WyYA+FouqkuRtk8bVHcYLqRUkK2n6dShEUGMuY1SzcAbBINvJYmQp+hhzK5m47AzCl463emXepYZQC/evytktG7yXxd3k8Ak+Qr7T4+G2VgJl4YrafTpIT6wowd+8u/SMSrrf/M41OhtLeBC4uDKjO3rYBQfVLTpEAgiX/9jxB80RtNMeCwgcieviAR5tlw2IlxVTEhxXbFeopcOZmfEuhVWqgBUfIakqsNCXkkubV0XS2l5G1vtTM8oNML0rP8PyKd4+0M5N6P/EQqFkHH93LCDD0IQBq9usm3MoJp0eT8N3m5gprI05drDh2xe/W6qnQfw3YXnjdvf2A=", "bob_encrypting_key": "026d1f4ce5b2474e0dae499d6737a8d987ed3c9ab1a55e00f57ad2d8e81fe9e9ac"}'
```

The received base64 encoded data would be deserialized to bytes.

Of course, validation isn't completely necessary as part of deserialization i.e. one option is that we can just take the bytes from deserialization, not do any validation, and pass them along to the `Porter.publish_treasure_map function` that ultimately passes them to the `TreasureMapPublisher` which takes bytes for treasure map. So we don't "need"  to validate, and therefore don't care about federated vs decentralized. If the bytes are invalid, presumably the publisher would fail.

Reply:
Potential fix via https://github.com/nucypher/nucypher/pull/2717/commits/f18db740a496fd013a2613363c4836e9224e1183.